src/index.ts

```diff
@@ -46,9 +90,26 @@ export function setCookie(
     if (typeof cookies === 'string') cookies = [cookies]
     if (typeof cookies === 'number') cookies = []
 
-    cookies.push(cookie.serialize(name, value, options))
-
-    ctx.res.setHeader('Set-Cookie', cookies)
+    const parsedCookies = setCookieParser.parse(cookies)
+
+    let cookiesToSet: string[] = []
+    parsedCookies.forEach((parsedCookie: Cookie) => {
+      if (!areCookiesEqual(parsedCookie, createCookie(name, value, options))) {
+        cookiesToSet.push(
+          cookie.serialize(parsedCookie.name, parsedCookie.value, {
```

Comment:
Couldn't we simply pass in `options` here?

Reply:
the `Cookie` interface is not compatible with the `CookieSerializeOptions` interface and the `options` (method argument) is not the same options as the `parsedCookie` options.